src/main/java/com/banking/app/bankingApp/service/transactions/TransactionManagementService.java

```diff
@@ -0,0 +1,91 @@
+            transaction.setPurpose(dbTransaction.getPurpose());
+            transaction.setCreatedAt(dbTransaction.getCreatedAt());
+            return transaction;
+        } else {
+
+            throw new IllegalStateException();
+        }
+    }
+
+    public Transaction getTransactionById(String id) {
+        Transaction transaction = new Transaction();
+        DBTransaction dbTransaction = transactionsDatabaseService.getDBTransactionById(id);
+        transaction.setId(dbTransaction.getId());
+        transaction.setAmount(dbTransaction.getAmount());
+        transaction.setPurpose(dbTransaction.getPurpose());
+        transaction.setSourceId(dbTransaction.getSourceAccount().getId());
+        transaction.setDestinationId(dbTransaction.getDestinationAccount().getId());
+        transaction.setCreatedAt(dbTransaction.getCreatedAt());
+        return transaction;
+    }
+
+    public List<Transaction> getAllTransactions(String id) {
+        List<Transaction> transactions = new ArrayList<>();
+        List<Transaction> transactions1 = new ArrayList<>();
+        List<DBTransaction> dbTransactions = transactionsDatabaseService.getAllDBTransactionsWhereSource(id);
+        List<DBTransaction> dbTransactions1 = transactionsDatabaseService.getAllDBTransactionsWhereDestination(id);
+        for (int i = 0; i < dbTransactions.size(); i++) {
+            Transaction transaction = new Transaction();
+            transaction.setId(dbTransactions.get(i).getId());
+            transaction.setAmount(dbTransactions.get(i).getAmount());
+            transaction.setPurpose(dbTransactions.get(i).getPurpose());
+            transaction.setSourceId(dbTransactions.get(i).getSourceAccount().getId());
+            transaction.setDestinationId(dbTransactions.get(i).getDestinationAccount().getId());
+            transaction.setCreatedAt(dbTransactions.get(i).getCreatedAt());
+            transactions1.add(transaction);
+        }
+        for (int i = 0; i < dbTransactions.size(); i++) {
+            Transaction transaction = new Transaction();
+            transaction.setId(dbTransactions1.get(i).getId());
+            transaction.setAmount(dbTransactions1.get(i).getAmount());
+            transaction.setPurpose(dbTransactions1.get(i).getPurpose());
+            transaction.setSourceId(dbTransactions1.get(i).getSourceAccount().getId());
+            transaction.setDestinationId(dbTransactions1.get(i).getDestinationAccount().getId());
+            transaction.setCreatedAt(dbTransactions1.get(i).getCreatedAt());
+            transactions.add(transaction);
+        }
+        for (int i = 0; i < dbTransactions.size() + dbTransactions1.size(); i++) {
+            transactions.add(transactions1.get(i));
+        }
+        return transactions;
+    }
+
+    public void updateTransaction(String id, UpdateTransaction updateTransaction) {
+        transactionsDatabaseService.updateDBTransaction(updateTransaction, id);
```

Comment:
In this code, you can update transactions amount even if there are not enough funds in the src account. Fix it